### ISSUE TITLE
[Question] fit-generator error with custom data generator

### ISSUE THREAD

**Issue by bgyooPtr** [keras-team/keras#14293]
I made a generator by inheriting the sequence class.
I know that class overrides the __getitem__ function to return data.
My model is not a classifier, and when I load the data I load the input and ground truth as well as some parameters needed to calculate the loss.

So, the return values of __getitem__, X and y, have the following shape.
**X** is a numpy object and its shape is (batch_size, 120, 120, 6).
**y** has 4 kinds of data including ground truth. (gt, a,b,c) -> {gt:(3,3,1), a:(3,3,1), b:(3,1), c:(4,4)}
Therefore, it is composed of a list.
The length of the list is batch_size.

After that, when I ran fit_generator, the following error occurred.
```
Please provide as model inputs either a single array or a list of arrays. You passed: target=None
```
What did I do wrong and how can I fix it?

**Comment by daniloguimaraes71** [comment]
can you share your code so I can reproduce it?

how are you outputting x and y?
have you tried using:

yield X, {'gt': gt, 'a':a, 'b':b, 'c':c}

this shape works for me.

**Comment by bgyooPtr** [comment]
@daniloguimaraes71 Please refer to the sample code below.
What I've found is that you can't configure multiple outputs. 
This is because when you call the generator's __getitem__ in the fit function, you may be confused with (input, targets, sample_weight).
But I don't know how to refer to the parameters P1 and P2 in the train_step function.
```
class DataGenerator(keras.utils.Sequence):
    def __init__(
        self, list_IDs, batch_size, dim, num_dim, camera_model, camera_image, crop, shuffle=True
    ):
        self.list_IDs = list_IDs
        self.batch_size = batch_size
        self.dim = dim
        self.num_dim = num_dim
        self.camera_model = camera_model
        self.camera_image = camera_image
        self.crop = crop
        self.shuffle = shuffle

        self.on_epoch_end()

    def on_epoch_end(self):
        "Updates indexes after each epoch"
        self.indexes = np.arange(len(self.list_IDs))
        if self.shuffle == True:
            np.random.shuffle(self.indexes)

    def __data_generation(self, list_IDs_temp):
        "Generates data containing batch_size samples"
        # Initialization
        X = np.empty((self.batch_size, *self.dim, self.num_dim * 2))
        labels = []
        batch_y = np.empty((self.batch_size, self.num_dim, 4), dtype=np.float32)
        batch_P1 = np.empty((self.batch_size, 4, 4), dtype=np.float32)
        batch_P2 = np.empty((self.batch_size, 3), dtype=np.float32)

        # Generatoe data
        for i, ID in enumerate(list_IDs_temp):
            """
            load the data and parameters
            """
            X[i,] = np.concatenate((img1, img2), axis=2)
            batch_y[i,] = y
            batch_P1[i,] = P1
            batch_P2[i,] = P2

        return (
            {"input_1": X, "P1": batch_P1, "P2": batch_P2},
            batch_y,
        )

    def __len__(self):
        "Denotes the number of batches per epoch"
        return int(np.floor(len(self.list_IDs) / self.batch_size))

    def __getitem__(self, index):
        "Generate one batch of data"
        # Generate indexes of the batch
        indexes = self.indexes[index * self.batch_size : (index + 1) * self.batch_size]

        # Find list of IDs
        list_IDs_temp = [self.list_IDs[k] for k in indexes]

        # Generate data
        return self.__data_generation(list_IDs_temp)
        # return (X), (y[0], y[1], y[2])

class CustomModel(tf.keras.Model):
    def train_step(self, data):
        X, y = data

        with tf.GradientTape() as tape:
            prob = self(X["input_1"], training=True)
            loss = loss_fn(prob, y, X["P1"], X["P2"])

        # Compute gradients
        trainable_vars = self.trainable_variables
        gradients = tape.gradient(loss, trainable_vars)

        # Update weights
        # self.optimizer.apply_gradients(zip(gradients, trainable_vars))
        opt.apply_gradients(zip(gradients, trainable_vars))

        # Compute our own metrics
        self.train_loss.update_state(loss)

        return {"train loss": self.train_loss.result(), "train mae": self.train_mae.result()}
```

**Comment by sushreebarsa** [comment]
@bgyooPtr Moving this issue to closed status as there has been no recent activity.In case you still face the error please create a new issue,we will get you the right help.Thanks!
